### PR TITLE
cloudflare_tiered_cache: add cf-tf support for new resource

### PR DESCRIPTION
**Pull request by foreseaz**
Add `cloudflare_tiered_cache` support to generate the resource of [tiered_cache](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/tiered_cache).

Tested locally with output:
```
make build && cd test && ../cf-terraforming generate --zone 0ebc4...2bc4fd6 --resource-type "cloudflare_tier
ed_cache"
resource "cloudflare_tiered_cache" "terraform_managed_resource_0ebc4...2bc4fd6" {
  cache_type = "smart"
  zone_id    = "0ebc4...2bc4fd6"
}
```

### CODE REVIEW

File: internal/app/cf-terraforming/cmd/generate.go [hunk 1]
```diff
@@ -1068,6 +1068,23 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+
+			resourceCount = 1
+			m, _ := json.Marshal(jsonPayload)
+			err = json.Unmarshal(m, &jsonStructData)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			jsonStructData[0].(map[string]interface{})["id"] = zoneID
+			jsonStructData[0].(map[string]interface{})["cache_type"] = tieredCache.Type.String()
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:1087]: why does this need remapping? isn't this the value in the schema already?

File: internal/app/cf-terraforming/cmd/generate.go [hunk 2]
```diff
@@ -1068,6 +1068,23 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+			jsonPayload = append(jsonPayload, tieredCache)
+
+			resourceCount = 1
+			m, _ := json.Marshal(jsonPayload)
+			err = json.Unmarshal(m, &jsonStructData)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			jsonStructData[0].(map[string]interface{})["id"] = zoneID
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:1086]: you shouldn't need to set the ID since it isn't used.

**Review by jacobbednarz — CHANGES_REQUESTED** [review]
thanks for the PR @foreseaz 🙇 i've added some comments in line but we'll also need to add test coverage for this. if you're looking for prior art, #496 is a good starting point.

File: internal/app/cf-terraforming/cmd/generate.go (hunk 1 above)
Comment (foreseaz) [reply internal/app/cf-terraforming/cmd/generate.go:1087]: Heya @jacobbednarz thanks for the feedback! Correct me if I'm wrong, I was trying to follow the schema defined in the provider here https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/tiered_cache

```
resource "cloudflare_tiered_cache" "example" {
  zone_id    = "0da42c8d2132a9ddaf714f9e7c920711"
  cache_type = "smart"
}
```
while the result from the API is type `TieredCache` as following, the `TieredCacheType` is enum so I suppose we need to call `String()` for translation. 
```
type TieredCache struct {
	Type         TieredCacheType
	LastModified time.Time
}
```
If there is a better way, kindly tell me and will update right away.

File: internal/app/cf-terraforming/cmd/generate.go (hunk 2 above)
Comment (foreseaz) [reply internal/app/cf-terraforming/cmd/generate.go:1086]: As mentioned in the last comment, I was following the provider example, if that's not needed I'll remove it.

**Comment by foreseaz** [comment]
@jacobbednarz Added tests, but when I was running `make test` locally, got errors as following:

```
?       github.com/cloudflare/cf-terraforming/cmd/cf-terraforming       [no test files]
=== RUN   TestGenerate_writeAttrLine
=== RUN   TestGenerate_writeAttrLine/value_is_nil
=== RUN   TestGenerate_writeAttrLine/value_is_string
=== RUN   TestGenerate_writeAttrLine/value_is_int
=== RUN   TestGenerate_writeAttrLine/value_is_float
=== RUN   TestGenerate_writeAttrLine/value_is_bool
=== RUN   TestGenerate_writeAttrLine/value_is_list_of_strings
=== RUN   TestGenerate_writeAttrLine/value_is_block_of_strings
--- PASS: TestGenerate_writeAttrLine (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_nil (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_string (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_int (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_float (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_bool (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_list_of_strings (0.00s)
    --- PASS: TestGenerate_writeAttrLine/value_is_block_of_strings (0.00s)
=== RUN   TestGenerate_ResourceNotSupported
time="2023-05-05T13:53:23-07:00" level=fatal msg="failed to detect provider installation"
FAIL    github.com/cloudflare/cf-terraforming/internal/app/cf-terraforming/cmd  17.905s
FAIL
make: *** [test] Error 1
```

Could you help on passing the test? Besides, let me know if there is anything off in the current commits.

**Comment by jacobbednarz** [comment]
you can either pass in `--terraform-install-path` to your tests or you can run these in an initialised terraform directory to get this going. i usually use the former via env vars in tests for simplicity.

**Comment by foreseaz** [comment]
@jacobbednarz Hi thanks for the last comment, and is there anything that needs working for this PR?

File: internal/app/cf-terraforming/cmd/generate.go (hunk 2 above)
Comment (jacobbednarz) [reply internal/app/cf-terraforming/cmd/generate.go:1086]: zone settings are a little weird and need this. i'll remove it.

File: internal/app/cf-terraforming/cmd/generate.go (hunk 1 above)
Comment (jacobbednarz) [reply internal/app/cf-terraforming/cmd/generate.go:1087]: that is all 👍 in the past we used to call `String()` automatically if it existed but i forgot we ditched it. cheers for the explanation.

**Pull request merged**